contracts/src/v0.8/keystone/CapabilityRegistry.sol

```diff
@@ -19,19 +22,36 @@ contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
     string name;
   }
 
-  struct Node {
+  struct NodeParams {
     /// @notice The id of the node operator that manages this node
-    uint256 nodeOperatorId;
+    uint32 nodeOperatorId;
+    /// @notice The signer address for application-layer message verification.
+    address signer;
     /// @notice This is an Ed25519 public key that is used to identify a node.
     /// This key is guaranteed to be unique in the CapabilityRegistry. It is
     /// used to identify a node in the the P2P network.
     bytes32 p2pId;
+    /// @notice The list of hashed capability IDs supported by the node
+    bytes32[] hashedCapabilityIds;
+  }
+
+  struct Node {
+    /// @notice The node's parameters
+    /// @notice The id of the node operator that manages this node
+    uint32 nodeOperatorId;
+    /// @notice The number of times the node's capability has been updated
+    uint32 configCount;
     /// @notice The signer address for application-layer message verification.
     address signer;
-    /// @notice The list of hashed capability IDs this node supports. This list is
-    /// never empty and all capabilities are guaranteed to exist in the
-    /// CapabilityRegistry.
-    bytes32[] supportedHashedCapabilityIds;
+    /// @notice This is an Ed25519 public key that is used to identify a node.
+    /// This key is guaranteed to be unique in the CapabilityRegistry. It is
+    /// used to identify a node in the the P2P network.
+    bytes32 p2pId;
+    /// @notice The node's supported capabilities
+    /// @dev This is stored as a map so that we can easily update to a set of
+    /// new capabilities by incrementing the configCount and creating a
+    /// new set of supported capability IDs
+    mapping(uint32 configCount => EnumerableSet.Bytes32Set capabilityId) supportedCapabilityIds;
```

Comment:
Is there a use case for seeing the capabilities supported by a node at a previous config count?

Reply:
Asking because we only seem to return the latest - is this just less gassy?

Reply:
Yes we basically need a quick way to track the latest supported capability IDs.  The alternative would have been to loop through all the capability IDs and remove them from the set one by one and adding the new capability IDs.  Adding the `configCount` allows us to just read the latest supported IDs.